src/content/components/BugList/BugList.js

```diff
@@ -93,32 +162,223 @@ export class BugList extends React.PureComponent {
 
   renderFilters() {
     return (
-      <EditorGroup>
-        {this.props.showResolvedOption ? (
-          <span>
-            <input
-              type="checkbox"
-              onChange={this.onCheckShowResolved}
-              checked={this.state.showResolved}
-            />{" "}
-            Show Resolved
-          </span>
-        ) : null}
-      </EditorGroup>
+      <div>
+        <EditorGroup>
+          {this.props.showResolvedOption ? (
+            <span>
+              <input
+                type="checkbox"
+                onChange={this.onCheckShowResolved}
+                checked={this.state.showResolved}
+              />{" "}
+              Show Resolved
+            </span>
+          ) : null}
+        </EditorGroup>
+      </div>
     );
   }
 
+  onEditDropdownSelect(selectedBugs, selection) {
+    this.setState({ editSelection: selection.value });
+    if (selection.value == 2) {
+      window.location.href = this.getBulkEditLink(selectedBugs);
+    }
+
+    if (selectedBugs.length == 1) {
+      const selectedBug = this.props.bugs.find(
+        bug => bug.id == parseInt(selectedBugs[0])
+      );
+
+      if (selection.value == 0) {
+        const newIteration =
+          selectedBug.cf_fx_iteration == "---"
+            ? this.context.iterations.getIteration().number
+            : this.context.iterations.getAdjacentIteration(
+                1,
+                null,
+                lookupIterations([selectedBug.cf_fx_iteration])
+                  .orderedVersionStrings[0]
+              ).number;
+        this.setState({ newIteration });
+      } else if (selection.value == 1) {
+        const newPriority =
+          selectedBug.priority == "--" ? "P1" : selectedBug.priority;
+        this.setState({ newPriority });
+      }
+    }
+  }
+
+  onUpdateIterationSelect(selection) {
+    this.setState({ newIteration: selection.value });
+  }
+
+  onUpdatePrioritySelect(selection) {
+    this.setState({ newPriority: selection.value });
+  }
+
+  updateBug(id, fields) {
+    this.setState({ bugUpdating: true });
```

Comment:
`bugUpdating: true` renders a text in place of the edit dropdown that says "Updating bug...". Right now, after updating the iteration or priority, it displays the "Updating bug..." text, the list refreshes, then after about 1-2 seconds the bug's iteration or priority is updated. I believe that this 1-2 second delay is because we first return cached bug data before returning bug data that is fetched from Bugzilla. Is this delay okay, or should I find an alternative to accessing cached bug data first?